tests/src/tracing/eventcounter/regression-25709.cs

```diff
@@ -0,0 +1,112 @@
+using Microsoft.Diagnostics.Tracing;
+#else
+using System.Diagnostics.Tracing;
+#endif
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+
+namespace EventCounterRegressionTests
+{
+
+    public class SimpleEventListener : EventListener
+    {        
+        private readonly EventLevel _level = EventLevel.Verbose;
+
+        public int MaxIncrement { get; private set; } = 0;
+
+        public SimpleEventListener()
+        {
+        }
+
+
+        protected override void OnEventSourceCreated(EventSource source)
+        {
+            if (source.Name.Equals("System.Runtime"))
+            {
+                Dictionary<string, string> refreshInterval = new Dictionary<string, string>();
+                refreshInterval.Add("EventCounterIntervalSec", "1");
+                EnableEvents(source, _level, (EventKeywords)(-1), refreshInterval);
+            }
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            int increment = 0;
+            bool isExceptionCounter = false;
+
+            for (int i = 0; i < eventData.Payload.Count; i++)
+            {
+                IDictionary<string, object> eventPayload = eventData.Payload[i] as IDictionary<string, object>;
+                if (eventPayload != null)
+                {
+                    foreach (KeyValuePair<string, object> payload in eventPayload)
+                    {
+                        if (payload.Key.Equals("Name") && payload.Value.ToString().Equals("exception-count"))
+                            isExceptionCounter = true;
+                        if (payload.Key.Equals("Increment"))
+                        {
+                            increment = Int32.Parse(payload.Value.ToString());
+                        }
+                    }
+                    if (isExceptionCounter)
+                    {
+                        if (MaxIncrement < increment)
+                        {
+                            MaxIncrement = increment;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public partial class TestEventCounter
+    {
+
+        public static void ThrowExceptionTask()
+        {
+            // This will throw an exception every 1000 ms
+            while (true)
+            {
+                Thread.Sleep(1000);
+                try
+                {
+                    throw new Exception("an exception");
+                }
+                catch
+                {}
+            }
+        }
+
+        public static int Main(string[] args)
+        {
+            Task exceptionTask = Task.Run(ThrowExceptionTask);
+            Thread.Sleep(5000);
+
+            // Create an EventListener.
+            using (SimpleEventListener myListener = new SimpleEventListener())
+            {
+                Thread.Sleep(5000);
+
+                if (myListener.MaxIncrement > 5)
```

Comment:
Please correct me if I'm getting something wrong but since the task throws an exception every second and we slept for 5 seconds after we created an EventListener to listen to it, I thought the theoretical maximum should be 5.

Reply:
During the 5 seconds we slept I'd expect the listener nominally to get 5 callbacks, each representing a one second sampling interval, and each reporting that one exception occurred. Larger numbers seem possible only if we have a bug or if the sampling interval is getting delayed excessively. Is that sort of delay what you were compensating for or something else?